lib/api/repository/auth_repository.dart

```diff
@@ -21,7 +21,7 @@ abstract class AuthRepository {
   });
 }
 
-@Singleton(as: AuthRepository)
+@LazySingleton(as: AuthRepository)
```

Comment:
I tried with this annotation but it didn't work on my IC 🤔 Looks weird 😄

Reply:
@hoangnguyen92dn Did you face this issue 👇  on your ic during the run Integration Test:

> flutter: *** DioError ***:
> flutter: uri: REST_API_ENDPOINT/api/v1/me
> flutter: DioError [unknown]: null
> Error: Invalid argument(s): No host specified in URI REST_API_ENDPOINT/api/v1/me

If not, You don't need to change `@Singleton` to `@LazySingleton`. Also, Have you run this script after changing the annotation `fvm flutter packages pub run build_runner build --delete-conflicting-outputs` before you rerun Integration Test 🤔  I still found the error on my side without `@LazySingleton` so I need to keep it 🙏

Reply:
> Or do you mean only AuthRepository we should revert back to @Singleton

@hoangnguyen92dn If you mean this ☝️   I think let's keep it as `@LazySingleton` to make it consistent with other repositories I think it would be better that we can ensure that the instance of the dependencies is properly initialized when the network requests so we don't face the `DioError` 🙇‍♀️

Reply:
@Wadeewee I faced the same exact issue, but even I tried with `@LazySingleton` the issue was still there 🤔 Can you please try with the Android build for confirmation? 😄

Reply:
Your right @hoangnguyen92dn 🙈  It doesn't work on `Android` anymore for both `@Singleton` and `@LazySingleton`  I'm checking this 🤔

Reply:
@Wadeewee On my repository, I added the `initDefault()` on each test case but the test is hanging on.

![image](https://github.com/Wadeewee/survey-flutter-ic-pooh/assets/6950766/69d9e193-875c-414b-b17e-5c649af14575)